datafusion/core/src/dataframe/parquet.rs

```diff
@@ -82,9 +83,19 @@ impl DataFrame {
                 .build()?
         };
 
+        let path = if file_type.get_ext() != DEFAULT_PARQUET_EXTENSION
+            && options.single_file_output
+        {
+            let mut path = path.to_owned();
+            path.push_str(SINGLE_FILE_EXTENSION);
+            path
+        } else {
+            path.to_owned()
+        };
```

Comment:
This part confuses me; If I'm not wrong `file_type.get_ext() != DEFAULT_PARQUET_EXTENSION` will always be true, because:

https://github.com/apache/datafusion/blob/602475fbac6b5fb78c28406775f8792f9ba5d70f/datafusion/datasource-parquet/src/file_format.rs#L157-L162